src/barchart.jl

```diff
@@ -1,6 +1,41 @@
 # Histogram (Bar) plots
 #
 # Vertical bars
+const ImPlotData = Union{Float32, Float64, Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64}
+
+const plot_bars_count = IdDict(Float32 => LibCImPlot.PlotBarsFloatPtrInt,
+                               Float64 => LibCImPlot.PlotBarsdoublePtrInt,
+                               Int8    => LibCImPlot.PlotBarsS8PtrInt,
+                               UInt8   => LibCImPlot.PlotBarsU8PtrInt,
+                               Int16   => LibCImPlot.PlotBarsS16PtrInt,
+                               UInt16  => LibCImPlot.PlotBarsU16PtrInt,
+                               Int32   => LibCImPlot.PlotBarsS32PtrInt,
+                               UInt32  => LibCImPlot.PlotBarsU32PtrInt,
+                               Int64   => LibCImPlot.PlotBarsS64PtrInt,
+                               UInt64  => LibCImPlot.PlotBarsU64PtrInt)
+
+const plot_bars_pointers = IdDict(Float32 => LibCImPlot.PlotBarsFloatPtrFloatPtr,
+                                  Float64 => LibCImPlot.PlotBarsdoublePtrdoublePtr,
+                                  Int8    => LibCImPlot.PlotBarsS8PtrS8Ptr,
+                                  UInt8   => LibCImPlot.PlotBarsU8PtrU8Ptr,
+                                  Int16   => LibCImPlot.PlotBarsS16PtrS16Ptr,
+                                  UInt16  => LibCImPlot.PlotBarsU16PtrU16Ptr,
+                                  Int32   => LibCImPlot.PlotBarsS32PtrS32Ptr,
+                                  UInt32  => LibCImPlot.PlotBarsU32PtrU32Ptr,
+                                  Int64   => LibCImPlot.PlotBarsS64PtrS64Ptr,
+                                  UInt64  => LibCImPlot.PlotBarsU64PtrU64Ptr)
+
+
+function PlotBars(values::Vector{T}; count::Integer = length(values),
+                  label_id::String = "", width = 0.67, shift = 0.0, offset::Integer = 0,
+                  stride::Integer = 1) where {T <: ImPlotData}
+
+    plot_bars_count[T](label_id, values, Cint(count), Float64(width),
+                                        Float64(shift), Cint(offset),
+                                        Cint(stride * sizeof(T)))
+end
+
+
```

Comment:
Here's one way we might solve consolidating the template functions. There might be a clever way to regex the function symbols and zip the types together, but this works for now

Reply:
Another alternative might be to give the wrapper generator a dictionary of plot names, and do regex there. In that case, the form of these functions would stay the same as shown here, but we could rip out the `IdDict`